plugins/dmxusb/src/dmxusbopenrx.cpp

```diff
@@ -0,0 +1,298 @@
+/*
+  Q Light Controller
+  dmxusbopenrx.cpp
+
+  Copyright (C) Heikki Junnila
+                Christopher Staite
+                Emmanuel Coirier
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0.txt
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include <QSettings>
+#include <QDebug>
+#include <math.h>
+#include <QTime>
+
+#include "dmxusbopenrx.h"
+#include "qlcmacros.h"
+
+#define DMX_MAB 16
+#define DMX_BREAK 110
+#define DMX_CHANNELS 512
+#define DEFAULT_OPEN_DMX_FREQUENCY    30  // crap
+#define SETTINGS_CHANNELS "dmxusbopenrx/channels"
+
+/****************************************************************************
+ * Initialization
+ ****************************************************************************/
+
+DMXUSBOpenRx::DMXUSBOpenRx(DMXInterface *interface,
+                                   quint32 inputLine, QObject* parent)
+    : QThread(parent)
+    , DMXUSBWidget(interface, 0, DEFAULT_OPEN_DMX_FREQUENCY)
+    , m_running(false)
+    , m_granularity(Unknown)
+    , m_reader_state(Idling)
+{
+    qDebug() << "Open RX contructor, line" << inputLine;
+
+    m_inputBaseLine = inputLine;
+    setOutputsNumber(0);
+    setInputsNumber(1);
+
+    m_inputLines[0].m_universeData = QByteArray();
+    m_inputLines[0].m_compareData = QByteArray();
+
+// on macOS, QtSerialPort cannot handle an OpenDMX device
+// so, unfortunately, we need to switch back to libftdi
+#if defined(Q_OS_OSX) && defined(QTSERIAL) && (defined(LIBFTDI1) || defined(LIBFTDI))
+    if (interface->type() == DMXInterface::QtSerial)
+        forceInterfaceDriver(DMXInterface::libFTDI);
+#endif
+    qDebug() << "Open RX contructor end";
+}
+
+DMXUSBOpenRx::~DMXUSBOpenRx()
+{
+    qDebug() << "Open RX destructor";
+    stop();
+    qDebug() << "Open RX stopped in destructor";
+}
+
+DMXUSBWidget::Type DMXUSBOpenRx::type() const
+{
+    return DMXUSBWidget::OpenRX;
+}
+
+/****************************************************************************
+ * Open & Close
+ ****************************************************************************/
+
+bool DMXUSBOpenRx::open(quint32 line, bool input)
+{
+    if (input == false) {
```

Comment:
There's a whole mix of parenthesis indentation in this file
Please use only one, same as the rest of the code. Parenthesis on new line. Thanks.